enhancements/console/pinned-resources.md

```diff
@@ -0,0 +1,179 @@
+
+![Default pinned resources](default-pinned-resources.png)
+
+## Motivation
+
+Some customers want to provide default pre-pinned resources for all users.
+
+### User Stories
+
+#### Story 1
+
+Customer wants to customize the developer perspective navigation by providing pre-pinned resources to users who do not have their navigation items customized.
+
+As a cluster admin, I want to provide default pre-pinned resources for all users.
+
+### Goals
+
+- Allow cluster admins to define default pre-pinned resources on the dev perspective navigation
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+We can extend the existing `operator.openshift.io/v1` / `Console CRD`.
+We can add `spec.customization.perspectives.pinnedResources` which lists default pre-pinned resources
+
+### Workflow Description
+
+To add default pre-pinned resources:
+
+```text
+├── spec
+│   ├── customization
+|       ├── perspectives
+|           ├── id
+|           ├── ...
+|           ├── pinnedResources
+|               └── group
+                └── version
+                └── kind
+└── ...
+```
+
+Each perspective conforms to the following schema :
+
+- `pinnedResources: authV1.GroupVersionKind[]` , defines a list of default pinned resources
+- `group: string` , defines the group of the resource type.
+- `version: string` , defines the version of the resource type.
+- `kind: string` , defines the kind of the resource type.
+
+If the `pinnedResources` list is empty, the default pinned resources will be shown.
+If the `pinnedResources` list is non-empty, only the listed items will be shown if the user hasn't already customized the navigation items.
+
+### API Extensions
+
+Extend the existing `operator.openshift.io/v1` / `Console CRD` by adding `spec.customization.perspectives.pinnedResources`
+
+So the example yaml should look like this:
+
+To set default pre-pinned resources
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata:
+  name: cluster
+  ...
+spec:
+  customization:
+    perspectives:
+    - id: 'dev'
+      ...
+      pinnedResources:
+      - group: apps
+        version: v1
+        kind: Deployment
+      - group: core
+        version: v1
+        kind: ConfigMap
+    - ...
+  ...
+```
+
+### Risks and Mitigations
+
+N/A
+
+### Drawbacks
+
+N/A
+
+## Design Details
+
+We will provide a configuration UI and a yaml snippet in the
+console to provide a list of available resources that can be pinned to the navigation.
+
+### Test Plan
+
+- If no pinnedResources are defined, then default pinned resources should be visible to users who do not have their navigation items customized
+- If pinnedResources is an empty list, then default pinned resources should be visible to users who do not have their navigation items customized
+- If the pinnedResources list is non-empty, then users who do not have their navigation customized should see the pinned resources defined in the customization
+
+### Graduation Criteria
+
+This will be released directly to GA.
```

Comment:
There is no benefit to introducing this as a tech preview first?

Reply:
I don't think this is required, but a good idea for customizations in general.

@debsmita1 we can check this with Serene when she is back from PTO.

Reply:
There is a movement within openshift, to improve stability, that all new features should be introduced first as tech previews, and then to GA. I guess it's not mandatory but something to consider as we are releasing in a more continuous fashion

Reply:
@serenamarie125  ^

Reply:
Resolved in this conversation https://coreos.slack.com/archives/CHGU4P8UU/p1670435710137509